tutorials/hw-encoding.md

```diff
@@ -0,0 +1,292 @@
+
+[GUI] [Err] [HWEncoder.cc:159] Could not initialize device /dev/nvidia3 for NVENC, skipping.
+[GUI] [Msg] Found known HW encoder: h264_omx
+[GUI] [Msg] Encoder h264_omx is not allowed, skipping.
+[GUI] [Msg] Found known HW encoder: h264_v4l2m2m
+[GUI] [Msg] Encoder h264_v4l2m2m is not allowed, skipping.
+[GUI] [Msg] Found known HW encoder: h264_vaapi
+[GUI] [Msg] Encoder h264_vaapi is not allowed, skipping.
+[GUI] [Msg] Found known HW encoder: nvenc
+[GUI] [Err] [Util.cc:58] ffmpeg [AVHWDeviceContext] Could not initialize the CUDA driver API
+
+[GUI] [Err] [HWEncoder.cc:159] Could not initialize device /dev/nvidia3 for NVENC, skipping.
+[GUI] [Msg] Found known HW encoder: nvenc_h264
+[GUI] [Err] [Util.cc:58] ffmpeg [AVHWDeviceContext] Could not initialize the CUDA driver API
+
+[GUI] [Err] [HWEncoder.cc:159] Could not initialize device /dev/nvidia3 for NVENC, skipping.
+[GUI] [Wrn] [HWEncoder.cc:384] No hardware-accelerated encoder found, falling back to software encoders
+[GUI] [Msg] Compatible SW encoder: libx264
+[GUI] [Msg] Using encoder libx264
+```
+
+# Examples
+
+Here are a few ready-made examples which might or might not work for you right-away.
+Just give them a try and dig deeper in the configuration if something is wrong.
+
+## Linux/Win + Intel GPU
+
+    IGN_VIDEO_ALLOWED_ENCODERS=QSV
+
+## Linux/Win + NVidia GPU
+
+    IGN_VIDEO_ALLOWED_ENCODERS=NVENC
+
+## Linux + Intel/NVidia GPU
+
+    IGN_VIDEO_ALLOWED_ENCODERS=VAAPI
+
+## Linux NVidia Multi-GPU machine
+
+    IGN_VIDEO_ALLOWED_ENCODERS=NVENC IGN_VIDEO_ENCODER_DEVICE=/dev/nvidia2
+
+# Pitfalls
+
+## NVEnc per-machine limit
+
+If you have a multi-GPU station with desktop-class (not server-class) GPUs, you will
+run into an artificial limitation from NVidia. You can only run 3 concurrent
+encoding sessions on one computer, no matter how many GPUs you have. The computer
+can even have 8 GPUs, but you will only be able to encode 3 videos at a time. The
+exact maxima of encoding sessions are described in
+[this NVidia support page](https://developer.nvidia.com/video-encode-and-decode-gpu-support-matrix-new).
+
+To make things worse, there is no API that would tell you the number of currently
+running NVEnc sessions. The only way to find out you're launching the fourth is
+trying to start encoding and getting a memory allocation error. This library
+catches this error and writes a lengthy description of what might have just happened
+(either really low memory or this artificial limit). Unfortunately, when you start
+the doomed fourth encoding session, all the three "legal" sessions will crash.
+This might be really troublesome on e.g. multi-user systems when you don't even know
+which jobs of the other users are using NVEnc.
+
+There is a workaround removing this artificial limit - patching the binary blob
+drivers using https://github.com/keylase/nvidia-patch . It is up to you if you
+can and want to do that. And there is no guarantee it will work forever.
```

Comment:
```suggestion
There is a workaround removing this artificial limit - patching the binary blob
drivers using https://github.com/keylase/nvidia-patch . This is an unofficial 
patch that is not supported by NVidia or the Ignition developers. It is up to 
you if you can and want to do that. And there is no guarantee it will work forever.
```